### PR TITLE
Add initial nRF91 support.

### PR DESCRIPTION
**Pull request by zyp**
This is a draft of initial support for nRF9160.

## Your checklist for this pull request

* [x] I've read the [Code of Conduct](https://github.com/blackmagic-debug/blackmagic/blob/main/CODE_OF_CONDUCT.md)
* [x] I've read the [guidelines for contributing](https://github.com/blackmagic-debug/blackmagic/blob/main/CONTRIBUTING.md) to this repository
* [x] It builds for hardware native (`make PROBE_HOST=native`)
* [x] It builds as BMDA (`make PROBE_HOST=hosted`)
* [ ] I've tested it to the best of my ability
* [x] My commit messages provide a useful short description of what the commits do

### CODE REVIEW

**Comment by mfiumara** [comment]
Hi @zyp, I've been looking for nrf91 support on the bmp for some time and it'd be great if this could be merged. Do you need help in testing it? I have a nrf91 thingy I could test this on.

**Comment by zyp** [comment]
This was originally blocked on a proper fix for handling the HNONSEC bit in CSW, which got merged in a82624c, so I think if I take out the workaround and rebase it to current main, it is mergeable.

@dragonmux Are there any other changes you'd like to see while I'm at it?

**Comment by mfiumara** [comment]
For me this does not seem to build @zyp, maybe I'm missing something:

```
 make PROBE_HOST=native
  BUILD   lib/stm32/f1
make[2]: Nothing to be done for `all'.
  BUILD   lib/stm32/f4
make[2]: Nothing to be done for `all'.
  BUILD   lib/lm4f
make[2]: Nothing to be done for `all'.
  LD      blackmagic.elf
/Applications/ArmGNUToolchain/12.3.rel1/arm-none-eabi/bin/../lib/gcc/arm-none-eabi/12.3.1/../../../../arm-none-eabi/bin/ld: blackmagic.elf section `.data' will not fit in region `rom'
/Applications/ArmGNUToolchain/12.3.rel1/arm-none-eabi/bin/../lib/gcc/arm-none-eabi/12.3.1/../../../../arm-none-eabi/bin/ld: region `rom' overflowed by 392 bytes
Memory region         Used Size  Region Size  %age Used
             rom:      131464 B       128 KB    100.30%
             ram:        3756 B        20 KB     18.34%
collect2: error: ld returned 1 exit status
make[1]: *** [blackmagic.elf] Error 1
make: *** [all] Error 2
```

**Comment by zyp** [comment]
Removed workaround, rebased to current main, fixed the minor API thing and the formatting the linter complained about.

Have only tested that scanning still works with BMDA, so it could do with some more testing, but apart from that I think this is ready to be merged.

**Comment by mfiumara** [comment]
> Removed workaround, rebased to current main, fixed the minor API thing and the formatting the linter complained about.
> 
> 
> 
> Have only tested that scanning still works with BMDA, so it could do with some more testing, but apart from that I think this is ready to be merged.

I can test this on my bmp v2 later today with an nrf91 thingy.

File: src/target/nrf91.c
Comment (dragonmux) [inline src/target/nrf91.c:86]: As this is all new code, and we're trying to reduce the amount of single letter variable names (it is a clang-tidy lint for a reason) could `t` be renamed `target` please. Similarly above and for `f` to be renamed `flash`

File: src/target/nrf91.c
Comment (dragonmux) [inline src/target/nrf91.c:90]: These braces are not required, please drop them

**Review by dragonmux — CHANGES_REQUESTED** [review]
This all looks good save for a couple of naming items and some errant braces. With them fixed, we'll be happy to merge 👍🏼

**Review by dragonmux — APPROVED** [review]
LGTM, we'll merge this once the builds complete. If there's something broken, a new PR with fixes can always be opened.

**Comment by dragonmux** [comment]
@zyp just noticed this was taken without ever getting a license notice added to the top of file - what license was this intended to be contributed under? We'll add it in a branch we're currently building as a fixup to the file.

**Comment by zyp** [comment]
Anything is okay for me, I'm happy for all contributions I do to be considered non-copyleft (MIT and/or BSD). However, the code is partially based on `nrf51.c` which does have a GPL header, so if you need a single license for the whole file, GPL restrictions would have to apply.

This goes for #2009 as well.

**Pull request merged**
